src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -148,9 +148,466 @@
+        }
+    }
+
+    for (int i = iStart; i <= iEnd; i++)
+    {
+        char ch = srcPattern[i];
+        switch (ch)
+        {
+            case CHAR_DIGIT:
+            case CHAR_ZERO:
+                if (!digitAdded)
+                {
+                    digitAdded = true;
+                    destPattern[index++] = 'n';
+                }
+                break;
+
+            case CHAR_CURRENCY:
+                if (!currencyAdded)
+                {
+                    currencyAdded = true;
+                    destPattern[index++] = 'C';
+                }
+                break;
+
+            case CHAR_SPACE:
+            case CHAR_NBSPACE:
+                if (!spaceAdded)
+                {
+                    spaceAdded = true;
+                    destPattern[index++] = ' ';
+                }
+                break;
+
+            case CHAR_MINUS:
+            case CHAR_OPENPAREN:
+            case CHAR_CLOSEPAREN:
+                minusAdded = true;
+                destPattern[index++] = (char)ch;
+                break;
+
+            case CHAR_PERCENT:
+                destPattern[index++] = '%';
+                break;
+        }
+    }
+
+    return destPattern;
+}
+
+/*
+Function:
+GetNumericPattern
+
+Determines the pattern from the decimalFormat and returns the matching pattern's
+index from patterns[].
+Returns index -1 if no pattern is found.
+*/
+static int GetNumericPatternNative(const char* pNumberFormat,
+                                   const char* patterns[],
+                                   int patternsCount,
+                                   int isNegative)
+{
+    const int INVALID_FORMAT = -1;
+    const int MAX_DOTNET_NUMERIC_PATTERN_LENGTH = 6; // example: "(C n)" plus terminator
+    char* normalizedPattern = NormalizeNumericPattern(pNumberFormat, isNegative);
+
+    if (!normalizedPattern)
+    {
+        return U_MEMORY_ALLOCATION_ERROR;
+    }
+
+    size_t normalizedPatternLength = strlen(normalizedPattern);
+
+    assert(normalizedPatternLength > 0);
+    assert(normalizedPatternLength < MAX_DOTNET_NUMERIC_PATTERN_LENGTH);
+
+    if (normalizedPatternLength == 0 || normalizedPatternLength >= MAX_DOTNET_NUMERIC_PATTERN_LENGTH)
+    {
+        free(normalizedPattern);
+        return INVALID_FORMAT;
+    }
+
+    for (int i = 0; i < patternsCount; i++)
+    {
+        if (strcmp(normalizedPattern, patterns[i]) == 0)
+        {
+            free(normalizedPattern);
+            return i;
+        }
+    }
+
+    assert(false); // should have found a valid pattern
+
+    free(normalizedPattern);
+    return INVALID_FORMAT;
+}
+
+int32_t GlobalizationNative_GetLocaleInfoIntNative(const char* localeName, LocaleNumberData localeNumberData)
+{
+    bool isSuccess = true;
+    int32_t value;
+    NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+    NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+
+    switch (localeNumberData)
```

Comment:
We don't have tests for most of these cases, do we?
From all the new functionalities I can see we are testing only `LocaleNumber_NegativeMonetaryNumberFormat` and `LocaleNumber_FirstWeekOfYear`.

Reply:
No not for all, need to add more test cases.